cli/mender-artifact/artifacts.go

```diff
@@ -326,7 +326,7 @@ func reconstructArtifactWriteData(ua *unpackedArtifact) (*awriter.WriteArtifactA
 }
 
 func repack(comp artifact.Compressor, ua *unpackedArtifact, to io.Writer, key []byte) error {
-	aWriter := awriter.NewWriter(to, comp)
+	aWriter := awriter.NewWriter(to, ua.ar.Compressor())
```

Comment:
Does this prevent repacking with a new compression type using `modify`? Because this is a quite important use case, if you're trying to deploy to a device and find that it's lacking lzma, for example.

Reply:
I see, hmm, then we will need a special case for this